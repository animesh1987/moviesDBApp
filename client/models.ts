export interface HeaderProps{
  genreSelected?: string,
  toggleType: Function,
  searchInputEnter: Function,
  onChangeInput: Function,
  onSearchClick: Function,
  searchInput?: string
}

export interface Movie {
  id: number,
  title: string,
  tagline: string,
  vote_average: number,
  vote_count: number,
  release_date: string,
  poster_path: string,
  overview: string,
  budget: number,
  revenue: number,
  runtime: number,
  genres: string[]
}

export interface State{
  genreSelected?: string,
  movies?: Movie[],
  searchInput?: string
}