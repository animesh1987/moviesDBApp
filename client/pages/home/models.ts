export interface Props{
  name: string,
  check: Function
}

export interface State{
  name: string,
  counter: number
}