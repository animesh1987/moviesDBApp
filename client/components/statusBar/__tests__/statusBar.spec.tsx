import * as React from 'react';
import { create } from 'react-test-renderer';
import { mount } from "enzyme";

import { StatusBar } from '../index';
import {Movie} from "../../../models";

describe('Sort By Test', () => {

  const sortBy = jest.fn();
  const sortedBy = 'release_date';
  const movie: Movie = {
    id: 1,
    title: 'Test',
    overview: 'Test overview',
    release_date: "20-04-04",
    runtime: 144,
    tagline: 'Test',
    genres: [
      'action'
    ],
  };;

  it('renders correctly', () => {
    const component = create(
      <StatusBar
        count={0}
        movieSelected={movie}
        sortBy={sortBy}
        sortedBy={sortedBy}
      />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const component = create(
      <StatusBar
        count={0}
        movieSelected={movie}
        sortBy={sortBy}
        sortedBy={sortedBy}
      />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('calls sort by function', () => {
    const component = mount(
      <StatusBar
        count={1}
        movieSelected={{}}
        sortBy={sortBy}
        sortedBy={sortedBy}
      />
    );
    component.find('span')
      .at(3).simulate('click');
    expect(sortBy).toBeCalled();
  });
});
