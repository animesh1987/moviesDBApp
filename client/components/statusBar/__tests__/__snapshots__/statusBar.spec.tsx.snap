// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort By Test renders correctly 1`] = `
<div
  className="status-bar"
>
  <div
    className="layout-row layout-align-center layout-align-space-between"
  >
    <span>
      Films by 
      action
       genre
    </span>
  </div>
</div>
`;

exports[`Sort By Test renders correctly 2`] = `
<div
  className="status-bar"
>
  <div
    className="layout-row layout-align-center layout-align-space-between"
  >
    <span>
      Films by 
      action
       genre
    </span>
  </div>
</div>
`;
